### DISCUSSION TITLE
Windows 端下通过Wox快速记录想法到flomo

### DISCUSSION THREAD

**Discussion by simonli** [flomoapp/3rd-party-tools#24] [分享创造]
快捷方式打开Wox的输入框，输入以下安装指令，即可安装Wox的flomo插件

1，wpm install Wox2Flomo
2，安装完毕后，请转到该插件的目录，打开“Wox.Plugin.Flomo.dll.config”，修改其中的ApiUrl为你自己的专属flomo API

 <add key="ApiUrl" value="https://flomoapp.com/iwh/MjAzMDQ/12cf31d14b187e6ff6c86a961cc0d81d/" /> 

http://www.wox.one/plugin/356

**Comment by hailibull** [comment]
You can help us to fix this issue by opening issue in: https://github.com/Wox-launcher/Wox/issues/new
1. upload log file: C:\Users\Administrator\AppData\Roaming\Wox\Logs\1.4.1196\2021-04-02.txt
2. copy below exception message
Error id: fdbb474461b4418481290878eeb3976d
Exception begin --------------------
  System.Exception: Update.exe not found, not a Squirrel-installed app?
  HResult: -2146233088
  Data: <System.Object> -> <>
  Source: Squirrel
  TargetAssembly: Squirrel, Version=1.9.1.0, Culture=neutral, PublicKeyToken=null
  TargetModule: Squirrel.dll
  TargetSite: System.String getUpdateExe()
  StackTrace: --------------------
   at Squirrel.UpdateManager.getUpdateExe() in C:\Users\ana\code\Squirrel\Squirrel.Windows\src\Squirrel\UpdateManager.cs:line 313
   at Squirrel.UpdateManager.RestartApp(String exeToStart, String arguments) in C:\Users\ana\code\Squirrel\Squirrel.Windows\src\Squirrel\UpdateManager.cs:line 196
   at Wox.Core.Plugin.PluginInstaller.Install(String path) in C:\projects\wox\Wox.Core\Plugin\PluginInstaller.cs:line 93
   at Wox.PublicAPIInstance.<>c__DisplayClass17_0.<InstallPlugin>b__0() in C:\projects\wox\Wox\PublicAPIInstance.cs:line 131
   at System.Windows.Threading.Dispatcher.Invoke(Action callback, DispatcherPriority priority, CancellationToken cancellationToken, TimeSpan timeout)
   at System.Windows.Threading.Dispatcher.Invoke(Action callback)
   at Wox.Plugin.PluginManagement.Main.<>c__DisplayClass12_0.<ResultForInstallPlugin>b__0(ActionContext c) in C:\projects\wox\Plugins\Wox.Plugin.PluginManagement\Main.cs:line 175
   at Wox.ViewModel.MainViewModel.<InitializeKeyCommands>b__17_7(Object index) in C:\projects\wox\Wox\ViewModel\MainViewModel.cs:line 209
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.UIElement.RaiseTrustedEvent(RoutedEventArgs args)
   at System.Windows.Input.InputManager.ProcessStagingArea()
   at System.Windows.Input.InputManager.ProcessInput(InputEventArgs input)
   at System.Windows.Input.InputProviderSite.ReportInput(InputReport inputReport)
   at System.Windows.Interop.HwndMouseInputProvider.ReportInput(IntPtr hwnd, InputMode mode, Int32 timestamp, RawMouseActions actions, Int32 x, Int32 y, Int32 wheel)
   at System.Windows.Interop.HwndMouseInputProvider.FilterMessage(IntPtr hwnd, WindowMessage msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at System.Windows.Interop.HwndSource.InputFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
Exception end ------------------------------------------------------------

## Runtime Info
* Command Line: "C:\Program Files (x86)\Wox-1.4.1196\Wox.exe" 
* Portable Mode: False
* Timestamp: 04/02/2021 13:15:58
* Wox version: 1.4.1196
* OS Version: Microsoft Windows NT 6.1.7601 Service Pack 1
* x64 OS: True
* x64 Process: True
* System Language: zh-CN
* Wox Language: zh-cn
* CLR Version: 4.0.30319.42000
* Installed .NET Framework: 
   * v2.0.50727 2.0.50727.5420 SP2
   * v3.0 3.0.30729.5420 SP2
   * v3.5 3.5.30729.5420 SP1
   * v4 Client 4.8.03761
   * v4 Full 4.8.03761
   * v4.0 Client 4.0.0.0
   * v4.8

## SDK Info
* Python Path: 
* Everything SDK Path: C:\Program Files (x86)\Wox-1.4.1196\Plugins\Wox.Plugin.Everything\EverythingSDK\x64\Everything.dll

## Assemblies - Wox.exe

* Wox, Version=1.4.1196.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Wox.exe)
* NLog, Version=4.0.0.0, Culture=neutral, PublicKeyToken=5120e14c03d0593c (C:\Program Files (x86)\Wox-1.4.1196\NLog.dll)
* Wox.Core, Version=1.4.1196.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Wox.Core.dll)
* Sentry, Version=2.1.1.0, Culture=neutral, PublicKeyToken=fba2ec45388e2af0 (C:\Program Files (x86)\Wox-1.4.1196\Sentry.dll)
* Sentry.Protocol, Version=2.1.1.0, Culture=neutral, PublicKeyToken=fba2ec45388e2af0 (C:\Program Files (x86)\Wox-1.4.1196\Sentry.Protocol.dll)
* System.Collections.Immutable, Version=1.2.3.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a (C:\Program Files (x86)\Wox-1.4.1196\System.Collections.Immutable.dll)
* Wox.Infrastructure, Version=1.4.1196.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Wox.Infrastructure.dll)
* System.Net.Http, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a (C:\Program Files (x86)\Wox-1.4.1196\System.Net.Http.dll)
* System.ValueTuple, Version=4.0.2.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51 (C:\Program Files (x86)\Wox-1.4.1196\System.ValueTuple.dll)
* Wox.Plugin, Version=1.4.1196.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Wox.Plugin.dll)
* Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed (C:\Program Files (x86)\Wox-1.4.1196\Newtonsoft.Json.dll)
* Squirrel, Version=1.9.1.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Squirrel.dll)
* Splat, Version=1.6.2.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Splat.dll)
* NuGet.Squirrel, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\NuGet.Squirrel.dll)
* Wox.Plugin.BrowserBookmark, Version=1.4.1196.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Plugins\Wox.Plugin.BrowserBookmark\Wox.Plugin.browserBookmark.dll)
* HelloWorldCSharp, Version=1.4.1196.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Plugins\HelloWorldCSharp\HelloWorldCSharp.dll)
* Wox.Plugin.Caculator, Version=1.4.1196.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Plugins\Wox.Plugin.Caculator\Wox.Plugin.Caculator.dll)
* Mages.Core, Version=1.6.0.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Plugins\Wox.Plugin.Caculator\Mages.Core.dll)
* Wox.Plugin.Color, Version=1.4.1196.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Plugins\Wox.Plugin.Color\Wox.Plugin.Color.dll)
* Wox.Plugin.ControlPanel, Version=1.4.1196.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Plugins\Wox.Plugin.ControlPanel\Wox.Plugin.ControlPanel.dll)
* Wox.Plugin.Folder, Version=1.4.1196.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Plugins\Wox.Plugin.Folder\Wox.Plugin.Folder.dll)
* Wox.Plugin.Everything, Version=1.4.1196.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Plugins\Wox.Plugin.Everything\Wox.Plugin.Everything.dll)
* Wox.Plugin.PluginIndicator, Version=1.4.1196.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Plugins\Wox.Plugin.PluginIndicator\Wox.Plugin.PluginIndicator.dll)
* Wox.Plugin.Program, Version=1.4.1196.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Plugins\Wox.Plugin.Program\Wox.Plugin.Program.dll)
* Wox.Plugin.PluginManagement, Version=1.4.1196.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Plugins\Wox.Plugin.PluginManagement\Wox.Plugin.PluginManagement.dll)
* Wox.Plugin.Shell, Version=1.4.1196.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Plugins\Wox.Plugin.Shell\Wox.Plugin.Shell.dll)
* WindowsInput, Version=1.0.4.0, Culture=neutral, PublicKeyToken=9b287f7dc5073cad (C:\Program Files (x86)\Wox-1.4.1196\WindowsInput.dll)
* Wox.Plugin.Sys, Version=1.4.1196.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Plugins\Wox.Plugin.Sys\Wox.Plugin.Sys.dll)
* Wox.Plugin.WebSearch, Version=1.4.1196.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Plugins\Wox.Plugin.WebSearch\Wox.Plugin.WebSearch.dll)
* Wox.Plugin.Url, Version=1.4.1196.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\Plugins\Wox.Plugin.Url\Wox.Plugin.Url.dll)
* Microsoft.Data.Sqlite, Version=3.1.3.0, Culture=neutral, PublicKeyToken=adb9793829ddae60 (C:\Program Files (x86)\Wox-1.4.1196\Plugins\Wox.Plugin.BrowserBookmark\Microsoft.Data.Sqlite.dll)
* NHotkey, Version=1.2.1.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\NHotkey.dll)
* NHotkey.Wpf, Version=1.2.1.0, Culture=neutral, PublicKeyToken=null (C:\Program Files (x86)\Wox-1.4.1196\NHotkey.Wpf.dll)
* CommandLine, Version=2.7.82.0, Culture=neutral, PublicKeyToken=5a870481e358d379 (C:\Program Files (x86)\Wox-1.4.1196\CommandLine.dll)
* ToolGood.Words, Version=3.0.1.2, Culture=neutral, PublicKeyToken=d330f4fd115e6143 (C:\Program Files (x86)\Wox-1.4.1196\ToolGood.Words.dll)
* Microsoft.WindowsAPICodePack.Shell, Version=1.1.4.0, Culture=neutral, PublicKeyToken=8985beaab7ea3f04 (C:\Program Files (x86)\Wox-1.4.1196\Microsoft.WindowsAPICodePack.Shell.dll)
* Microsoft.WindowsAPICodePack, Version=1.1.4.0, Culture=neutral, PublicKeyToken=8985beaab7ea3f04 (C:\Program Files (x86)\Wox-1.4.1196\Microsoft.WindowsAPICodePack.dll)
* ICSharpCode.SharpZipLib, Version=1.2.0.246, Culture=neutral, PublicKeyToken=1b03e6acf1164f73 (C:\Program Files (x86)\Wox-1.4.1196\ICSharpCode.SharpZipLib.dll)
* Sentry.PlatformAbstractions, Version=1.1.0.0, Culture=neutral, PublicKeyToken=fba2ec45388e2af0 (C:\Program Files (x86)\Wox-1.4.1196\Sentry.PlatformAbstractions.dll)
* System.Runtime.InteropServices.RuntimeInformation, Version=4.0.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a (C:\Program Files (x86)\Wox-1.4.1196\System.Runtime.InteropServices.RuntimeInformation.dll)
* mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (C:\Windows\Microsoft.NET\Framework64\v4.0.30319\mscorlib.dll)
* PresentationFramework, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35 (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\PresentationFramework\v4.0_4.0.0.0__31bf3856ad364e35\PresentationFramework.dll)
* WindowsBase, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35 (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\WindowsBase\v4.0_4.0.0.0__31bf3856ad364e35\WindowsBase.dll)
* System.Core, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\System.Core\v4.0_4.0.0.0__b77a5c561934e089\System.Core.dll)
* System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\System\v4.0_4.0.0.0__b77a5c561934e089\System.dll)
* PresentationCore, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35 (C:\Windows\Microsoft.Net\assembly\GAC_64\PresentationCore\v4.0_4.0.0.0__31bf3856ad364e35\PresentationCore.dll)
* System.Xaml, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\System.Xaml\v4.0_4.0.0.0__b77a5c561934e089\System.Xaml.dll)
* System.Configuration, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\System.Configuration\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Configuration.dll)
* System.Xml, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\System.Xml\v4.0_4.0.0.0__b77a5c561934e089\System.Xml.dll)
* netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51 (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\netstandard\v4.0_2.0.0.0__cc7b13ffcd2ddd51\netstandard.dll)
* System.Runtime.Remoting, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\System.Runtime.Remoting\v4.0_4.0.0.0__b77a5c561934e089\System.Runtime.Remoting.dll)
* System.Data, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (C:\Windows\Microsoft.Net\assembly\GAC_64\System.Data\v4.0_4.0.0.0__b77a5c561934e089\System.Data.dll)
* System.ServiceModel, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\System.ServiceModel\v4.0_4.0.0.0__b77a5c561934e089\System.ServiceModel.dll)
* System.Runtime.Serialization, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\System.Runtime.Serialization\v4.0_4.0.0.0__b77a5c561934e089\System.Runtime.Serialization.dll)
* System.Numerics, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\System.Numerics\v4.0_4.0.0.0__b77a5c561934e089\System.Numerics.dll)
* System.Xml.Linq, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\System.Xml.Linq\v4.0_4.0.0.0__b77a5c561934e089\System.Xml.Linq.dll)
* System.Drawing, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\System.Drawing\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Drawing.dll)
* System.Runtime.Caching, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\System.Runtime.Caching\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Runtime.Caching.dll)
* PresentationFramework.Aero, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35 (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\PresentationFramework.Aero\v4.0_4.0.0.0__31bf3856ad364e35\PresentationFramework.Aero.dll)
* WindowsFormsIntegration, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35 (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\WindowsFormsIntegration\v4.0_4.0.0.0__31bf3856ad364e35\WindowsFormsIntegration.dll)
* System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\System.Windows.Forms\v4.0_4.0.0.0__b77a5c561934e089\System.Windows.Forms.dll)
* UIAutomationTypes, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35 (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\UIAutomationTypes\v4.0_4.0.0.0__31bf3856ad364e35\UIAutomationTypes.dll)
* UIAutomationProvider, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35 (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\UIAutomationProvider\v4.0_4.0.0.0__31bf3856ad364e35\UIAutomationProvider.dll)
* PresentationFramework-SystemData, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\PresentationFramework-SystemData\v4.0_4.0.0.0__b77a5c561934e089\PresentationFramework-SystemData.dll)
* PresentationFramework-SystemXml, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\PresentationFramework-SystemXml\v4.0_4.0.0.0__b77a5c561934e089\PresentationFramework-SystemXml.dll)
* PresentationFramework-SystemXmlLinq, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\PresentationFramework-SystemXmlLinq\v4.0_4.0.0.0__b77a5c561934e089\PresentationFramework-SystemXmlLinq.dll)
* Accessibility, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a (C:\Windows\Microsoft.Net\assembly\GAC_MSIL\Accessibility\v4.0_4.0.0.0__b03f5f7f11d50a3a\Accessibility.dll)
* System.Web, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a (C:\Windows\Microsoft.Net\assembly\GAC_64\System.Web\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Web.dll)

**Reply by simonli** [reply]
Update.exe not found, not a Squirrel-installed app?

**Reply by chrvip** [reply]
You should restart wox after change the api path in file Wox.Plugin.Flomo.dll.config, or you will got a failure when send msg to flomo.
